File: agents/src/audit.py
import os
import httpx

_AUDIT_URL = os.environ.get("AUDIT_SERVICE_URL", "http://audit-logger:4000")

def log(
    agent: str,
    action_type: str,
    summary: str,
    *,
    target_schema: str | None = None,
    target_table: str | None = None,
    node_id: str | None = None,
    mode_active: str = "normal",
    metadata: dict | None = None,
) -> None:
    """Fire-and-forget audit log. Never raises — audit failure must not break agent logic."""
    try:
        httpx.post(
            f"{_AUDIT_URL}/log",
            json={
                "agent": agent,
                "action_type": action_type,
                "target_schema": target_schema,
                "target_table": target_table,
                "node_id": node_id,
                "summary": summary,
                "mode_active": mode_active,
                "metadata": metadata or {},
            },
            timeout=3,
        )
    except Exception as e:
        print(f"[audit] failed to log: {e}")

File: scrapers/src/audit.js
// Shared audit helper — all scrapers call this, never write to audit.log directly
const AUDIT_URL = process.env.AUDIT_SERVICE_URL ?? 'http://audit-logger:4000';

async function log({ action_type, target_schema, target_table, node_id, summary, metadata = {} }) {
  try {
    await fetch(`${AUDIT_URL}/log`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({
        agent: 'scraper',
        action_type,
        target_schema: target_schema ?? 'property_deals',
        target_table,
        node_id: node_id ? String(node_id) : null,
        summary,
        mode_active: 'normal',
        metadata,
      }),
    });
  } catch (err) {
    // Audit failure must never crash a scrape job
    console.warn('[audit] failed to log:', err.message);
  }
}

module.exports = { log };
